src/js/apps/patients/worklist/worklist_state.js

```diff
@@ -107,35 +110,66 @@ export default Backbone.Model.extend({
       [dateType]: this.formatDateRange(relativeRange, unit, dateFormat),
     };
   },
+  getDefaultStatesFilter() {
+    const isDoneOnly = this.id === 'done-last-thirty-days';
+    const { done, notDone } = this.states.groupByDone();
+
+    return isDoneOnly ? done : notDone;
+  },
+  setDefaultFilterStates() {
+    const defaultStatesFilter = this.getDefaultStatesFilter();
+    this.set({ filters: { states: defaultStatesFilter.map('id') } });
+  },
+  getSelectedStates() {
+    const defaultStatesFilterIds = this.getDefaultStatesFilter().map('id');
+    const selectedStates = this.getFilters().states;
+
+    // remove any invalid state ids (i.e. ids that don't belong to any default states)
+    return filter(selectedStates, id => contains(defaultStatesFilterIds, id)).join();
+  },
+  getFiltersCount() {
+    const filtersState = this.get('filters');
+    const defaultStatesFilter = this.getDefaultStatesFilter();
+
+    const customFiltersState = omit(filtersState, 'states');
+    const customFiltersSize = size(omit(customFiltersState, function(value) {
+      return isNull(value);
+    }));
+
+    const statesFilterSize = size(filtersState.states);
+    const defaultStatesSize = size(defaultStatesFilter.map('id'));
+
+    const statesFilterCount = statesFilterSize !== defaultStatesSize ? 1 : 0;
```

Comment:
I think this check should probably be `_.isEqual(filtersState.states.sort(), defaultStatesFilter.map('id').sort()) ?...`. seems like size checks would return false positives

Reply:
Updated in last push.